include/seqan3/search/detail/search.hpp

```diff
@@ -80,45 +80,41 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
     };
 
     // choose mode
-    if constexpr (search_traits_t::search_best_hits)
+    if constexpr (search_traits_t::search_best_hits ||
+                  search_traits_t::search_all_best_hits ||
+                  search_traits_t::search_strata_hits)
     {
         detail::search_param max_error2{max_error};
         max_error2.total = 0;
         while (internal_hits.empty() && max_error2.total <= max_error.total)
         {
-            detail::search_algo<true>(index, query, max_error2, internal_delegate);
+            /*
+            * If you only want the best hit (search_traits_t::search_best_hits), you stop after finding the first hit,
```

Comment:
Could you make it a line comment? 💅 
It looks like a doxygen entry gone wrong. 🙈